src/scripts/onerepStatsAlert.js

```diff
@@ -0,0 +1,80 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import Sentry from "@sentry/nextjs";
+
+const SENTRY_SLUG = "cron-onerep-stats-alerts";
+
+const MAX_MANUAL_SCANS = parseInt(process.env.MAX_MANUAL_SCANS) || 0;
+const MAX_INITIAL_SCANS = parseInt(process.env.MAX_INITIAL_SCANS) || 0;
+const MAX_PROFILES_ACTIVATED =
+  parseInt(process.env.MAX_PROFILES_ACTIVATED) || 0;
+const MAX_PROFILES_CREATED = parseInt(process.env.MAX_PROFILES_CREATED) || 0;
+
+Sentry.init({
+  environment: process.env.APP_ENV,
+  dsn: process.env.SENTRY_DSN,
+  tracesSampleRate: 1.0,
+});
+
+const checkInId = Sentry.captureCheckIn({
+  monitorSlug: SENTRY_SLUG,
+  status: "in_progress",
+});
+
+/**
+ * Fetch the latest usage statistics from OneRep's API.
+ *
+ * @see https://docs.onerep.com/#tag/Statistics
+ */
+export async function checkStats() {
+  const scans = await (await onerepFetch("/stats/scans")).json();
+  const profiles = await (await onerepFetch("/stats/profiles")).json();
+
+  for (const alert of [
+    ["Free scans", scans.manual, MAX_MANUAL_SCANS],
+    ["Paid scans", scans.initial, MAX_INITIAL_SCANS],
+    ["Profiles activated", profiles.activated, MAX_PROFILES_ACTIVATED],
+    ["Profiles created", profiles.created, MAX_PROFILES_CREATED],
+  ]) {
+    const [name, current, max] = alert;
+    if (current >= max) {
+      const msg = `Alert: OneRep scans over limit for ${name}. Current: ${current}, max: ${max}`;
+      console.error(msg);
+      Sentry.captureMessage(msg);
```

Comment:
Q: will sentry be able to tie this error message to a user ID or anything useful? Or will we just get _x_ reports of "Alert: OneRep scans over limit for Free scans. Current: 100, max: 100"?

Or are these limits system wide and not for a specific user?

Reply:
No I don't think it's very user-specific - it's just a quota on a global resource.

I'm sending it to Sentry now out of convenience, all I really want is to be able to generate alerts if we ever see this. We're going to send any of these events to Grafana.

Reply:
Oh also to be clear, getting any of these alerts means flipping these features off :) So just one is enough to start alerting on, we'll start getting it regularly since it's run from cron once any of these are elevated.